packages/jaeger-ui/src/model/ddg/transformDdgData.test.js

```diff
@@ -170,7 +171,7 @@ describe('transform ddg data', () => {
     const simpleModel = transformDdgData(wrap([simplePath, longSimplePath]), focalPayloadElem);
     const reverseModel = transformDdgData(wrap([longSimplePath, simplePath]), focalPayloadElem);
 
-    expect(reverseModel).not.toEqual(simpleModel);
+    expect(_isEqual(reverseModel, simpleModel)).toBe(false);
```

Comment:
for my education, why is `lodash/isEqual` necessary vs. the built-in assertion?

Reply:
also, this feels like a very weak test that can pass for many different reasons, not at all related to the specific condition being tested.

Reply:
But I am still confused, like why are these two objects giving false on equality. Because they are 100% equal. I tried using a [file diff utility ](https://codebeautify.org/file-diff) for that, and their output is `No Changes Found`

Reply:
This is the comparison for `console.log` of both of these objects

![image](https://github.com/jaegertracing/jaeger-ui/assets/94157520/0af79a3e-bbdc-40f6-b4f9-9c4ba06e3364)

Reply:
my understanding of this test is that the underlying data is slightly different, maybe in how items are ordered, but the hash computation is expected to be the same.

Reply:
So, I think this expect statement is redundant since we are anyway comparing their pointers, and hash values
```js
expect(reverseModel).not.toBe(simpleModel);
expect(reverseModel.hash).toBe(simpleModel.hash);
```
Should I remove them?

Reply:
comparing pointers is pointless, it's irrelevant. If you have two identical objects with different pointers, it's natural that their hashes will be the same. The point of this test was to ensure that structurally different objects still have the same hashes.

Reply:
Yeah, I just ran the test to see that too. I suggest removing the assertions against the models themselves, only leave the hash comparisons. We know from L170-171 that the models are constructed from structurally different data, there is no need to validate that again.